mne/io/nirx/nirx.py

```diff
@@ -459,6 +459,9 @@ def __init__(self, fname, saturated, preload=False, verbose=None):
         annot = Annotations(onset, duration, description, ch_names=ch_names)
         self.set_annotations(annot)
 
+        sort_idx = np.argsort(self.ch_names)
+        self.pick(picks=sort_idx)
```

Comment:
Hum. This will make a copy of the data buffer but more importantly this is not consistent with the other readers which read the channels in the order they appear in the file on disk. I prefer to avoid doing magic in readers. Relevant functions down the road should deal with files with channels present in different orders, eg combine_evoked grand_average etc.

Reply:
> This will make a copy of the data buffer

If `preload=False` it won't because we can pick without loaded data now. This means people who really need to save this 2x memory can do `read_raw_nirx(...).load_data()` instead. But really NIRS data tends to be pretty small so I don't think this will be an issue in practice.

> more importantly this is not consistent with the other readers which read the channels in the order they appear in the file on disk.

I agree with this idea in principle, but it will require a big refactoring of our existing code, which assumes and requires that the frequency pairings are essentially properly "interlaced" to work correctly.

So to me I would like to add a comment `# TODO: Remove this reordering`, which would give us some months to properly craft order-agnostic code, which I think will be a bigger undertaking involving coordination between MNE-Python and MNE-NIRS and changes in probably 5-10 public functions...

Reply:
@rob-luke I realized this is problematic. Let's say we have this arrangement with the more-or-less standard "channel numbering":
```
S1 -- 1 -- D1 -- 2 -- S2
|          |           |
3          4           5
|          |           |
D3 -- 6 -- S3 -- 7 -- D4
```
The native file will be written as `['S1_D1 760', 'S1_D2_850`, 'S2_D1_760', 'S2_D1_850', 'S1_D3_760', 'S1_D3_850', ...]`, i.e., a pair ordering of `[S1_D1, S2_D1, S1_D3, S3_D1, S2_D4, S3_D3, S3_D4]`, and people will naturally expect these pairs to be associated with channel numbers 1 through 7. But this `pick(argsort(ch_names))` will result in a reordered pair ordering of `[S1_D1, S1_D3, S2_D1, S2_D4, S3_D1, S3_D3, S3_D4]`, so channels 1-7 from before now show up in the order `[1, 3, 2, 5, 4, 6, 7]`.

Rather than this `pick(argsort(...))` it seems like it would make sense to reorder the channels such that the original "logical channel" ordering is preserved, but such that pairs are still immediately adjacent. We could start by making sure that the first `N` channels are all fNIRS channels, and that they are paired. Non-fNIRS channels go after that in channels `N+1:`. Sound good?

Eventually like we said before we could/should relax this by making the channel logic smarter everywhere, but that's a bigger undertaking. All we need to do to fix this I think is `.pick(a_better_order)` where `a_better_order` comes from a few lines of pairing logic.

Reply:
I disagree that there is a standard logical channel ordering. Even for the trivial example which you illustrate I would assign the channel numbers differently to you.

You have assigned
- Channel 1: s1 d1
- Channel 2: s2 d1
- Channel 3: s1 d3

I would have thought the logical ordering was 
- Channel 1: s1 d1
- Channel 2: s1 d3
- Channel 3: s2 d1

I also don't think that the channel number has any meaning. And it's the pair naming that should be used. 

That's why I went for a simple sort. It will always be the same and can be easily described.

Reply:
Sure it can differ by system or setup, but to me the acquisition system sets the "standard order" when it writes data to disk in some order. So the idea is just to respect the order as written to disk as much as possible until we can remove any ordering restrictions whatsoever.

FWIW we used to have this order problem with MEEG data, too, and removing it made the code cleaner in the end. So I'm looking forward to removing the paired-order restriction for fNIRS, as the cleanest long term solution is just to keep the order each manufacturer uses in the first place, or a user does with reorder_channels, etc. Currently reorder_chaanels is problematic for fNIRS and it shouldn't be!